### PR TITLE
XRENDERING-518: The macro content descriptor should specify if it can be edited inline

### PR DESCRIPTION
**Pull request by surli**
* Fix the macro message to put the unchanged content at the right
place

### CODE REVIEW

File: xwiki-rendering-macros/xwiki-rendering-macro-box/src/main/java/org/xwiki/rendering/internal/macro/box/DefaultBoxMacro.java [hunk 1]
```diff
@@ -46,6 +47,12 @@
 @Singleton
 public class DefaultBoxMacro<P extends BoxMacroParameters> extends AbstractBoxMacro<P>
 {
+    /**
+     * All the macro that needs to display an unchanged content div around the content.
+     */
+    private static final List<String> ACCEPTED_MACRO_UNCHANGED_CONTENT = Arrays.asList("box", "info", "error",
+        "warning", "success");
```
Comment (vmassol) [inline xwiki-rendering-macros/xwiki-rendering-macro-box/src/main/java/org/xwiki/rendering/internal/macro/box/DefaultBoxMacro.java]: That's really bad and a no go for me. It means that the BoxMacro knows about the message macros which is really bad.

Comment (vmassol) [reply xwiki-rendering-macros/xwiki-rendering-macro-box/src/main/java/org/xwiki/rendering/internal/macro/box/DefaultBoxMacro.java]: Just to be clear: it's a no go unless we will fix it soon, i.e missing a todo comment to explain what should/will be done :)

Comment (surli) [reply xwiki-rendering-macros/xwiki-rendering-macro-box/src/main/java/org/xwiki/rendering/internal/macro/box/DefaultBoxMacro.java]: Just added the comments

Comment (tmortagne) [reply xwiki-rendering-macros/xwiki-rendering-macro-box/src/main/java/org/xwiki/rendering/internal/macro/box/DefaultBoxMacro.java]: We should definitely not have that in 10.10 final.

**Pull request merged**